autogpt/agent/agent.py

```diff
@@ -327,7 +343,12 @@ def get_self_feedback(self, thoughts: dict, llm_model: str) -> str:
         plan = thoughts.get("plan", "")
         thought = thoughts.get("thoughts", "")
         feedback_thoughts = thought + reasoning + plan
-        return create_chat_completion(
+
+        messages = {"role": "user", "content": feedback_prompt + feedback_thoughts}
+
+        feedback = create_chat_completion(
             [{"role": "user", "content": feedback_prompt + feedback_thoughts}],
             llm_model,
         )
```

Comment:
oh you don't need to change the signature, 
1- just keep returning the feedback.
2- log the messages WITHIN the get_self_feedback method as well
3- log the feedback WITHIN the get_self_feedback method as well

And make sure your FIRST log the messages (prompt_supervisor_feedback.json) before you log the feedback (supervisor_feedback.txt)

Reply:
Yes, I did it before and tests were broken. Check it know, I did reset it.